jme3-core/src/main/java/com/jme3/app/DetailedProfiler.java

```diff
@@ -57,6 +57,9 @@ public void appStep(AppStep step) {
             for (StatLine statLine : data.values()) {
                 for (Iterator<Integer> i = statLine.taskIds.iterator(); i.hasNext(); ) {
                     int id = i.next();
+                    //FixMe
+                    //throws with app.settings.putBoolean("GraphicsDebug", true);
+                    //reason: query has not started/ended or is already running
```

Comment:
Can you elaborate? Is this caused by the PR?

Reply:
No its not caused by PR, the line below renderer.isTaskResultAvailable(id)
issues 
`gl.glGetQueryObjectiv(taskId, GL.GL_QUERY_RESULT_AVAILABLE)`
This method is not allowed to be called unless 
`gl.glBeginQuery(GL.GL_TIME_ELAPSED, taskId);` and 
`gl.glEndQuery(GL.GL_TIME_ELAPSED);`
are called. In other words it asks if result is ready without first starting and ending a query thus causing a glerror.

To resolve: This file will have to be rewritten with the use of the newly introduced QueryObject not only to fix this issue but also to not cause issues with user defined Queries.

Reply:
Ok. Then if you can plase open an issue for this.